client/consensus/babe/src/lib.rs

```diff
@@ -1542,6 +1543,47 @@ where
 					log::Level::Info
 				};
 
+				if viable_epoch.as_ref().end_slot() <= slot {
```

Comment:
```suggestion
				if viable_epoch.as_ref().end_slot() < slot {
```

Shouldn't it be this way?

Reply:
The range is closed for the start and open for the end (like iterator ranges); 
in other words the `end_slot` value doesn't belong to the range.

```
[----------------)
^                ^
start           end
```

Reply:
Yeah, @andresilva already told me this. Nevertheless, ty @davxy.